scripts/mk.js

```diff
@@ -14742,6 +14931,59 @@ function touche_cbleue_aux(iX,iY, oBox) {
 	return false;
 }
 
+function powCpuDodge(oKart) {
+	let kartZ = oKart.z;
+
+	if (!oKart.jumped) {
+		let n = Math.pow(2, -(iDificulty - 4)) * 0.5;
+		if (Math.random() < n) {
+			return;
+		}
+		else {
+			kartZ = 1 - n + (Math.random() * 1) * (iDificulty / 12);
+			if (kartZ < 0) kartZ = 0.1;
+			else if (kartZ > 1) kartZ = 1;
+		}
+
+		oKart.z = kartZ;
+		oKart.heightinc = kartZ / 2;
+		oKart.jumped = true;
+	}
+}
+
+function powEffect(i, oKart) {
+	let fullHit = oKart.z == 0;
+	let spinPower = fullHit ? 62 : Math.floor(Math.abs((oKart.z - 1.2) * 25) + 20);
+	if (spinPower % 2 === 1) spinPower++;
+
+	loseUsingItems(oKart);
+	dropCurrentItem(oKart);
+	oKart.spin(spinPower);
+	oKart.champi = 0;
+	delete oKart.champiType;
+	stopDrifting(i);
+
+	if (fullHit) {
+		aKarts[i].z = 1;
+		aKarts[i].heightinc = 0.5;
+		aKarts[i].jumped = true;
+		handleItemHit(oKart, "pow");
+	}
+
+	if (course === "BB" && fullHit)
+		popBalloon(oKart);
+}
+
+function playPow(i, oKart, oKartOwner, countdown, countstate) {
+	if (countstate === 5 && countdown === 10)
+		if (oKart != oKartOwner && oKart.cpu && oKart.z < 1.2 && !oKart.tourne && !friendlyFire(oKart, oKartOwner))
+			powCpuDodge(oKart);
+	
+	if (countstate === 6 && countdown === 1)
+		if (oKart != oKartOwner && !oKart.protect && oKart.z < 1.2 && !friendlyFire(oKart, oKartOwner))
```

Comment:
Maybe here we should add a condition for online mode: only run this code if you control the affected player. This way we reduce the risks of lag effects, the player only loses his items when he's hit from his point of view, regardless of point of view of others (same issue with the puddle I told you the other day)
```suggestion
		if (oKart != oKartOwner && !oKart.protect && oKart.z < 1.2 && !friendlyFire(oKart, oKartOwner) && (!isOnline || !i || kart.controller == identifiant))
```

Reply:
So that's how you fix that!! Thank you very much for solving this issue, that was something I noticed while testing and I had no idea how to fix it 🙏